Recent changes in LatencyReporter is causing an index out of bounds

It seems that the most recent commit https://github.com/dacapobench/dacapobench/commit/9b195d53687bc027026f1139010be568a696b1d5 caused a bug.

Running Cassandra with `java -Djava.security.manager=allow -jar dacapo-evaluation-git-9b195d53.jar cassandra -s small -n 1` results in:
```
Failed to invoke DaCapo LatencyReporter.start(): java.lang.reflect.InvocationTargetException
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at site.ycsb.Client.dacapoRequestStart(Client.java:457)
        at site.ycsb.ClientThread.run(ClientThread.java:126)
        at java.base/java.lang.Thread.run(Thread.java:1570)
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 2100 out of bounds for length 2000
        at org.dacapo.harness.LatencyReporter.startIdx(LatencyReporter.java:112)
        at org.dacapo.harness.LatencyReporter.start(LatencyReporter.java:105)
        at org.dacapo.harness.LatencyReporter.start(LatencyReporter.java:98)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        ... 4 more
```

Going back one commit (`3c443273`) to restore the recent changes in `LatencyReporter.java` eliminates the crash and runs as expected. 

Crash is triggered in a small workload by cassandra and tomcat.

--- a/benchmarks/harness/src/org/dacapo/harness/LatencyReporter.java
+++ b/benchmarks/harness/src/org/dacapo/harness/LatencyReporter.java
@@ -102,7 +102,9 @@ public class LatencyReporter {
       next_idx = inc();
     }
     idx = next_idx++;
-    startIdx(idx, id);
+    if (idx < txbegin.length) {
+      startIdx(idx, id);
+    }
     return idx;
   }
   private static void startIdx(int index, int threadID) {

